### PR TITLE
Fix spark2-sql-plugin

### PR DESCRIPTION
**Pull request by GaryShen2008**
Update spark2-sql-plugin based on 26a6a96
contributes to https://github.com/NVIDIA/spark-rapids/issues/6544, leaving open to check once more later

### CODE REVIEW

**Comment by tgravescs** [comment]
so we don't want to update the spark2 code until after we have frozen and changes are done, unless you plan on doing ti again?

**Comment by tgravescs** [comment]
I checked out your branch and just ran the diff script but it failed, please take a look and make sure that has a clean run.

```
$ ./rundiffspark2.sh
Done running Diffs of spark2 files
Using temporary directory: /tmp/spark2diff-cifBjZ6ENz
```

**Comment by GaryShen2008** [comment]
> so we don't want to update the spark2 code until after we have frozen and changes are done, unless you plan on doing ti again?

Yes, I plan to do it again after the code freeze. That's why I wrote the base commit id so that I just need to compare from that commit id.

**Comment by GaryShen2008** [comment]
> HiveProviderImpl

ok, it's strange that my local running passed. Let me check.
$ ./rundiffspark2.sh
Done running Diffs of spark2 files
Using temporary directory: /var/folders/hx/nyg3np7d43jdl5bv98080vt80000gq/T/spark2diff-XXXXXXXXXX.lbGWlLAX
Done running Diffs of spark2.x files

> I checked out your branch and just ran the diff script but it failed, please take a look and make sure that has a clean run.
> 
> ```
> $ ./rundiffspark2.sh
> Done running Diffs of spark2 files
> Using temporary directory: /tmp/spark2diff-cifBjZ6ENz
> *** spark2diffs/HiveProviderImpl.diff   2022-09-20 14:18:07.150326902 -0500
> ```

**Comment by GaryShen2008** [comment]
The problem is caused by a different version of "diff".
I ran it on my Mac, it's using diff v2.8.1. But on ubuntu, it's using diff v3.6.
The output of diff is different between these 2 versions.
I'll update the spark2diff by using diff v3.6.

File: spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuBatchUtils.scala [hunk 1]
```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2020-2021, NVIDIA CORPORATION.
```
Comment (tgravescs) [inline spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuBatchUtils.scala:2]: copyright should be 2022

File: spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimGpuOverrides.scala [hunk 2]
```diff
@@ -90,7 +71,7 @@ object ShimGpuOverrides extends Logging {
      GpuOverrides.exec[FileSourceScanExec](
         "Reading data from files, often from Hive tables",
         ExecChecks((TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.STRUCT + TypeSig.MAP +
-          TypeSig.ARRAY + TypeSig.DECIMAL_128).nested(), TypeSig.all),
+            TypeSig.ARRAY + TypeSig.BINARY + TypeSig.DECIMAL_128).nested(), TypeSig.all),
```
Comment (tgravescs) [inline spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimGpuOverrides.scala:74]: spacing 2 spces

File: spark2-sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveOverrides.scala [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022, NVIDIA CORPORATION.
+ * Copyright (c) 2020-2022, NVIDIA CORPORATION.
```
Comment (tgravescs) [inline spark2-sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveOverrides.scala:2]: leave 2020 off

File: spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala [hunk 4]
```diff
@@ -17,14 +17,11 @@
 package com.nvidia.spark.rapids
 
 import java.time.ZoneId
-
```
Comment (tgravescs) [inline spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala:20]: put new lines back

File: spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimGpuOverrides.scala (hunk 2 above)
Comment (GaryShen2008) [reply spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimGpuOverrides.scala]: Updated.

File: spark2-sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveOverrides.scala (hunk 3 above)
Comment (GaryShen2008) [reply spark2-sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveOverrides.scala]: Updated.

File: spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala (hunk 4 above)
Comment (GaryShen2008) [reply spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala]: Updated in the latest commit.

**Comment by tgravescs** [comment]
this is still in draft so is the plan to leave this open until end or are you going to merge and then put up a new pr?

**Comment by GaryShen2008** [comment]
My plan is to update this PR with the latest PR today, then we can merge it. If there's any related PR later, I'll create new PR to update again.

**Comment by GaryShen2008** [comment]
build

**Review by tgravescs — APPROVED** [review]
Please don't force push, it makes reviewing what changed impossible

**Comment by GaryShen2008** [comment]
> Please don't force push, it makes reviewing what changed impossible

ok, it seems force push required after rebasing to the latest code.
Is there any way to avoid the force push in that case?

**Comment by tgravescs** [comment]
if force push was required then the branch wasn't merged right or you weren't up to date when you started... you should do a merge and not a rebase for instance.  Not really sure without knowing exact steps you took.

**Pull request merged**